packages/jellyfish-api-core/__tests__/category/loan/takeLoan.test.ts

```diff
@@ -0,0 +1,216 @@
+    })
+    await expect(promise).rejects.toThrow(RpcApiError)
+    await expect(promise).rejects.toThrow('Invalid Defi token: INVALID')
+  })
+
+  it('should not takeLoan on incorrect auth', async () => {
+    const promise = tGroup.get(1).rpc.loan.takeLoan({
+      vaultId: vaultId,
+      amounts: '30@TSLA'
+    })
+    await expect(promise).rejects.toThrow(RpcApiError)
+    await expect(promise).rejects.toThrow(`Incorrect authorization for ${vaultAddress}`)
+  })
+
+  it('should not takeLoan while exceed vault collateralization ratio', async () => {
+    const promise = tGroup.get(0).rpc.loan.takeLoan({
+      vaultId: vaultId,
+      amounts: '300000@TSLA'
+    })
+    await expect(promise).rejects.toThrow(RpcApiError)
+    await expect(promise).rejects.toThrow('Vault does not have enough collateralization ratio defined by loan scheme')
+  })
+
+  it('should not takeLoan on mintable:false token', async () => {
+    await tGroup.get(0).container.call('updateloantoken', ['TSLA', { mintable: false }])
+    await tGroup.get(0).generate(1)
+    await tGroup.waitForSync()
+
+    const promise = tGroup.get(0).rpc.loan.takeLoan({
+      vaultId: vaultId,
+      amounts: '30@TSLA'
+    })
+    await expect(promise).rejects.toThrow(RpcApiError)
+    await expect(promise).rejects.toThrow('Loan cannot be taken on token with id (2) as "mintable" is currently false')
+
+    await tGroup.get(0).container.call('updateloantoken', ['TSLA', { mintable: true }])
+    await tGroup.get(0).generate(1)
+    await tGroup.waitForSync()
+  })
+
+  it('should takeLoan', async () => {
+    const vaultBefore = await tGroup.get(0).container.call('getvault', [vaultId])
+    const vaultBeforeTSLAAcc = vaultBefore.loanAmount.length > 0
+      ? vaultBefore.loanAmount.find((amt: string) => amt.split('@')[1] === 'TSLA')
+      : undefined
+    const vaultBeforeTSLAAmt = vaultBeforeTSLAAcc !== undefined ? Number(vaultBeforeTSLAAcc.split('@')[0]) : 0
+
+    const txid = await tGroup.get(0).rpc.loan.takeLoan({
+      vaultId: vaultId,
+      amounts: '30@TSLA'
+    })
+    expect(typeof txid).toStrictEqual('string')
+    await tGroup.get(0).generate(1)
+    await tGroup.waitForSync()
+
+    const vaultAfter = await tGroup.get(0).container.call('getvault', [vaultId])
+    console.log('vaultAfter: ', vaultAfter)
+    const vaultAfterTSLAAcc = vaultAfter.loanAmount.find((amt: string) => amt.split('@')[1] === 'TSLA')
+    const vaultAfterTSLAAmt = Number(vaultAfterTSLAAcc.split('@')[0])
+    expect(vaultAfterTSLAAmt - vaultBeforeTSLAAmt).toStrictEqual(30.0000171)
+  })
+
+  it('should takeLoan with utxos', async () => {
+    const vaultBefore = await tGroup.get(0).container.call('getvault', [vaultId])
+    const vaultBeforeTSLAAcc = vaultBefore.loanAmount.length > 0
+      ? vaultBefore.loanAmount.find((amt: string) => amt.split('@')[1] === 'TSLA')
+      : undefined
+    const vaultBeforeTSLAAmt = vaultBeforeTSLAAcc !== undefined ? Number(vaultBeforeTSLAAcc.split('@')[0]) : 0
```

Comment:
The is vaultBeforeTSLAAmt value after https://github.com/DeFiCh/jellyfish/pull/661/files#diff-0be0c027884e19e265cf69090bcedf997f57f33eac3a37de8505bd3aa575de6fR177 executed.

You probably should create separate vault so every test unit is independent